examples/source_separation/utils/dataset/librimix.py

```diff
@@ -0,0 +1,73 @@
+from pathlib import Path
+from typing import Union, Tuple, List
+
+import torch
+from torch.utils.data import Dataset
+
+import torchaudio
+
+SampleType = Tuple[int, torch.Tensor, List[torch.Tensor]]
+
+
+class LibriMix(Dataset):
+    r"""Create the LibriMix dataset.
+
+    Args:
+        root (str or Path): the path to the directory where the dataset is stored.
+        num_speakers (int, optional): The number of speakers, which determines the directories
+            to traverse. The Dataset will traverse ``s1`` to ``sN`` directories to collect
+            N source audios. (Default: 2)
+        sample_rate (int, optional): sample rate of audio files. If any of the audio has a
+            different sample rate, raises ``ValueError``. (Default: 8000)
```

Comment:
I know this is coming from my implementation of WSJ dataset, but if we are moving this implementation to the library, then maybe we should change this behavior to resample to the target sample rate, or let the client code resample it. 

The reason why I took this approach back then was that `torchaudio.transform.Resample` was so slow, and WSJ0Mix dataset generates sample rate of 8k and 16k Hz. So I assumed that just sticking to 8k and 16k Hz should be fine.

I think it's fine now.

Reply:
Yep LibriMix also has both 8k and 16k Hz versions. Most of the papers use the 8k Hz dataset. So I think it's fine.